hoot-core/src/main/cpp/hoot/core/criterion/HighwayCriterion.h

```diff
@@ -35,19 +37,26 @@ namespace hoot
 /**
  * A criterion that will either keep or remove road matches.
  */
-class HighwayCriterion : public ElementCriterion
+class HighwayCriterion : public ElementCriterion, public ConstOsmMapConsumer
```

Comment:
Being a `ConstOsmMapConsumer` derived class allows this class to use the `BuildingCriterion` class to make sure that the ways in question aren't buildings or related to buildings.